Scripts/Runtime/Entities/TaskSystem/TaskStream/TaskStream.cs

```diff
@@ -7,30 +9,88 @@ namespace Anvil.Unity.DOTS.Entities.Tasks
 
-        internal override bool IsCancellable
+
+        internal sealed override bool IsCancellable
         {
-            get => false;
+            get => (Flags & TaskStreamFlags.IsCancellable) != 0;
         }
 
-        internal TaskStream()
+        internal sealed override bool IsDataStreamAResolveTarget
+        {
+            get => (Flags & TaskStreamFlags.IsResolveTarget) != 0;
+        }
+        
+        internal TaskStream(TaskStreamFlags flags)
         {
-            //TODO: #71 - This is weird with no dispose but #71 will fix it.
+            Flags = flags;
+
+            Type type = GetType();
+
+            //TODO: Extract to Anvil-CSharp Util method -Used in AbstractJobConfig as well
+            m_TypeString = type.IsGenericType
+                ? $"{type.Name[..^2]}<{type.GenericTypeArguments[0].Name}>"
+                : type.Name;
+            
             DataStream = new EntityProxyDataStream<TInstance>();
+
+            if (IsCancellable)
+            {
+                PendingCancelDataStream = new EntityProxyDataStream<TInstance>();
+            }
+        }
+
+        protected override void DisposeSelf()
+        {
+            DataStream.Dispose();
+            PendingCancelDataStream?.Dispose();
+            base.DisposeSelf();
+        }
+        
+        public override string ToString()
+        {
+            return m_TypeString;
         }
 
         internal sealed override AbstractEntityProxyDataStream GetDataStream()
         {
             return DataStream;
         }
 
-        internal override AbstractEntityProxyDataStream GetPendingCancelDataStream()
+        internal sealed override AbstractEntityProxyDataStream GetPendingCancelDataStream()
+        {
+            Debug_EnsureCancellable();
+            return PendingCancelDataStream;
+        }
+        
+        //*************************************************************************************************************
+        // SAFETY
+        //*************************************************************************************************************
+
+        [Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+        private void Debug_EnsureCancellable()
         {
-            throw new NotSupportedException($"Tried to get Pending Cancel Data Stream on {this} but it doesn't exists. Check {IsCancellable} first!");
+            if (!IsCancellable)
+            {
+                throw new NotSupportedException($"Tried to get Pending Cancel Data Stream on {this} but it doesn't exists. Check {IsCancellable} first!");
+            }
         }
     }
+    
+    public abstract class AbstractTaskStream : AbstractAnvilBase
+    {
+        internal abstract bool IsCancellable { get; }
+        internal abstract bool IsDataStreamAResolveTarget { get; }
+        internal abstract AbstractEntityProxyDataStream GetDataStream();
+        internal abstract AbstractEntityProxyDataStream GetPendingCancelDataStream();
+    }
```

Comment:
is this worth just making an internal interface that extends `IAnvilDisposable`?

If not, since this is a public type it probably makes more sense in its own file

Reply:
I think the problem with the interface is that even though the interface itself is internal, the methods it defines have to be public. So anyone who has reference to a `TaskStream` can see the methods. 

I moved this to its own file.

Reply:
Ah yea, good point. All good